pymc4/distributions/state_functions.py

```diff
@@ -0,0 +1,161 @@
+import tensorflow as tf
+import tensorflow_probability as tfp
+from tensorflow_probability.python.mcmc.internal import util as mcmc_util
+
+tfd = tfp.distributions
+
+__all__ = ["categorical_uniform_fn", "bernoulli_fn", "gaussian_round_fn", "poisson_fn"]
+
+
+def categorical_uniform_fn(event_shape, scale=1.0, name=None):
+    """Returns a callable that samples new proposal from Categorical distribution with uniform probabilites
+    Args:
+       scale: a `Tensor` or Python `list` of `Tensor`s of any shapes and `dtypes`
+         controlling the upper and lower bound of the uniform proposal
+         distribution.
+       name: Python `str` name prefixed to Ops created by this function.
+           Default value: 'categorical_uniform_fn'.
+    Returns:
+        categorical_uniform_fn: A callable accepting a Python `list` of `Tensor`s
+         representing the state parts of the `current_state` and an `int`
+         representing the random seed used to generate the proposal. The callable
+         returns the same-type `list` of `Tensor`s as the input and represents the
+         proposal for the RWM algorithm.
+    """
+
+    def _fn(state_parts, seed):
+        with tf.name_scope(name or "categorical_uniform_fn"):
+            scales = scale if mcmc_util.is_list_like(scale) else [scale]
+            if len(scales) == 1:
+                scales *= len(state_parts)
+            if len(state_parts) != len(scales):
+                raise ValueError("`scale` must broadcast with `state_parts`")
+            deltas = [
```

Comment:
Using a tfd.Uniform will be more efficient to draw uniform categorical sample.

Reply:
I see, ok fixing this too.

Reply:
@junpenglao did you mean `tfd.Categorical`?